### ISSUE TITLE
API: Passwords with special characters aren't accepted (401)

### ISSUE THREAD

**Issue by AlfredSK** [friendica/friendica#6355]
### Expected behavior
If we support special characters in passwords authentication must also work via API.

### Actual behavior
If the password contains special characters you get an error 401.

### Steps to reproduce the problem
A user of my node reported that he cannot use any app because he always gets authentication errors. After setting a new password w/o special characters it works.

Characters used while getting the 401: `) ( ! : .`

### Friendica version you encountered the problem
2018.12-rc

### Friendica source (git, zip)
git

### PHP version
7.2.10

### SQL version
10.1.34 MariaDB

**Comment by MrPetovan** [comment]
Do you know which kind of special characters?

**Comment by MrPetovan** [comment]
I did a little digging, and the problem is that there's no standard to encode/decode user and passwords with the HTTP Basic Authentication headers. Basically, every browser encodes as it sees fit (often UTF-8), and every web server decode as they see fit (often ISO-8859-1).

The worst part is that even characters that are present in ISO-8859-1 (latin1) like `ü` end up mangled because the browser encodes them in UTF-8 before base64-encoding them.

Bottom line is: don't use non-ASCII characters in password.

This leaves a-z, A-Z, 0-9, and the following punctuation marks excluding the space (we trim inputs) and colon (used to separate user and password in the header): ``!"#$%&'()*+,-./;<=>?@[\]^_`{\}~``

All there is to do I guess is check entered password match this rule and make it clear in the register page.

**Comment by AlfredSK** [comment]
Yes, I know which special characters. Look at the issue text, section 'Steps to reproduce'. ;-)

**Comment by MrPetovan** [comment]
Closed via #6356
